skimage/measure/_regionprops.py

```diff
@@ -631,8 +631,8 @@ def solidity(self):
     @property
     def centroid_weighted(self):
         ctr = self.centroid_weighted_local
-        return tuple(idx + slc.start
-                     for idx, slc in zip(ctr, self.slice))
+        return tuple(idx + slc.start * spc
+                     for idx, slc, spc in zip(ctr, self.slice, self._spacing))
```

Comment:
I'm probably not grasping all the bits and pieces properly yet but I'm confused by something. I checked how the spacing is taken into account in the non-weighted case in 
https://github.com/scikit-image/scikit-image/blob/2f9b82ce95e95c558ffe68bced6225c9a4560e31/skimage/measure/_regionprops.py#L433-L438

(called by `self.centroid`). There it looks like the spacing is multiplied with the sum of index and offset / `slice.start`, while in this fix it's only multiplied with `slice.start`.  Should it be 
```suggestion
        return tuple((idx + slc.start) * spc
                     for idx, slc, spc in zip(ctr, self.slice, self._spacing))
```
instead?

Reply:
The non-weighted centroid is computed in a much different way, as it's much simpler. It computes the centroid without taking into account spacing, and applies all the corrections at the very end.

When computing the weighted centroid we start from the local (to the prop bbox) centroid, which is already corrected for spacing, and we just need to apply a translation to the coordinate system of the original (full size) image. If we apply the correction again as suggested this would yield wrong coordinates, as the local centroid would be corrected twice.

More in details: the centroid weighted local uses the weighted moments

https://github.com/scikit-image/scikit-image/blob/2f9b82ce95e95c558ffe68bced6225c9a4560e31/skimage/measure/_regionprops.py#L638-L660

and the weighted moments just use the moments which take spacing into account (see line 261 in particular):

https://github.com/scikit-image/scikit-image/blob/2f9b82ce95e95c558ffe68bced6225c9a4560e31/skimage/measure/_moments.py#L154-L269

Does this clear the concern.

Reply:
Yes, I think I'm following now. Thanks for the detailed explanation and the contribution in total! :D